internal/models/user.go

```diff
@@ -81,6 +82,21 @@ func GetUserById(userId uint) (*User, error) {
 	return user, err
 }
 
+func GetUsersFromEmails(emails []string) (map[string]*User, error) {
+	var users []*User
+	userMap := make(map[string]*User)
+
+	err := db.
+		Where("email IN ?", emails).
```

Comment:
Are emails normalized in db or will this create issues with capitalization? A `normalizedEmail` field in db may be good otherwise? For this use-case and when looking up by email in the future.

Reply:
You're right it might create issues ; the current `email` field in db should contain only lowercase emails, and the fetched emails in commits would be set as lowercase.

Reply:
I think so. I also think it's best to save the original email also, and not just a normalized one--hence why I would recommend a normalizedEmail field rather than normalizing the current email field. Emails can technically be case sensitive and it could technically screw with future email notification/sending functionality if the original is not kept.

Reply:
Your way of doing it makes good enough sense for the scale of this project. Same with not bothering to care about migrating old emails in db to lowercase. Maybe in a few years it's a different story. :)